pyartcd/pyartcd/pipelines/build_rhcos.py

```diff
@@ -58,184 +52,72 @@ def __init__(self, runtime: Runtime, new_build: bool, ignore_running: bool, vers
-        builds = []
-        for job in ("build", "build-arch", "release"):
-            response = self.request_session.get(
-                f"{JENKINS_BASE_URL}/job/{job}/api/json?tree=builds[number,description,result,actions[parameters[name,value]]]",
-            )
-            builds_info = response.json()["builds"]
-            for b in builds_info:
-                # build is still running when it has no status
-                if b["result"] is None:
-                    build = dict(**b, job=job, parameters=self.build_parameters(b), url=self.build_url(job, b["number"]))
-                    builds.append(build)
-
-        return [b for b in builds if b["parameters"].get("STREAM") == self.stream]
-
-    @property
-    def stream(self):
-        if self._stream:
-            return self._stream
-
-        # doozer --quiet -g openshift-4.14 config:read-group urls.rhcos_release_base.multi --default ''
-        # https://releases-rhcos-art.apps.ocp-virt.prod.psi.redhat.com/storage/prod/streams/4.14-9.2/builds
-        cmd = [
-            "doozer",
-            "--quiet",
-            "--group", f'openshift-{self.version}',
-            "config:read-group",
-            "urls.rhcos_release_base.multi",
-            "--default",
-            "''"
-        ]
-        result = subprocess.run(cmd, stdout=PIPE, stderr=PIPE, check=False, universal_newlines=True)
-        if result.returncode != 0:
-            raise IOError(f"Command {cmd} returned {result.returncode}: stdout={result.stdout}, stderr={result.stderr}")
-        match = re.search(r'streams/(.*)/builds', result.stdout)
-        if match:
-            self._stream = match[1]
-        else:
-            self._stream = self.version
-        return self._stream
+        """Check if there are any existing builds for the given job with given version. Returns builds dicts."""
+        if self.ignore_running:
+            return []
+        response = self.request_session.get(
+            f"{JENKINS_BASE_URL}/job/{self.job}/api/json?tree=builds[number,description,url,result,actions[parameters[name,value]]]",
+        )
+        builds_info = response.json()["builds"]
+        return [{"result": None, "description": build["description"], "url": build["url"]}
+                for build in builds_info
+                if build["result"] is None
+                and any(
+                    param["name"] == "STREAM" and param["value"] == self._stream  # check build parameter with given stream
+                    for action in build["actions"]
+                    if "parameters" in action
+                    for param in action["parameters"]
+        )]
+
+    def get_stream(self):
+        """Get rhcos job's stream parameter value, for 4.12+ it looks like 4.x-9.x for 4.12 is just 4.12"""
+        group_response = self.request_session.get(f"{OCP_REPO_RAW}/openshift-{self.version}/group.yml")
```

Comment:
Think raw.githubuserccontent.com content will read from cache, and there is no guarantee that it is up to date. Which is fine almost always, but just when we change this, we want this to be right. 

Soon, we'll want to be able to trigger this for `4.20-9.6` and `4.20-10.0`. Maybe the rhel major/minor should also be a job input? Not sure about the best place to look for that info, and if this is the PR to think about that.

Reply:
we just need to el version vars, the current design is to read from config, of course it can be set from parameter, but I don't think the el version will frequently change. (if we use parameter to set the stream, then maybe rhcos vars and config in group it not needed anymore, eg after this change the multi config is not needed anymore)

Reply:
if we use the parameter to set stream, then the parent job that triggers it also needs to decide the value, the logic seem still needed

Reply:
Current logic is to run `doozer`, which runs a clone and gets the current value. 
Proposed logic is to ask GitHub cache about the value. This is wrong for a while when this value gets changed. This is a type of defect that is very time consuming to debug. Let's not have this bug.